packages/coinjoin/src/types/coordinator.ts

```diff
@@ -0,0 +1,173 @@
+}
+
+export interface IssuerParameter {
+    cw: string;
+    i: string;
+}
+
+export interface IssuanceData {
+    realAmountCredentials: ZeroCredentials;
+    realVsizeCredentials: ZeroCredentials;
+    zeroAmountCredentials: RealCredentials;
+    zeroVsizeCredentials: RealCredentials;
+}
+
+export interface ConfirmationData {
+    realAmountCredentials: RealCredentials;
+    realVsizeCredentials: RealCredentials;
+    zeroAmountCredentials: RealCredentials;
+    zeroVsizeCredentials: RealCredentials;
+}
+
+export interface RegistrationData {
+    aliceId: string;
+    amountCredentials: RealCredentials;
+    vsizeCredentials: RealCredentials;
+    isPayingZeroCoordinationFee: boolean;
+}
+
+export interface CoordinatorFeeRate {
+    rate: number;
+    plebsDontPayThreshold: number;
+}
+
+export interface CoinjoinRoundCreatedEvent {
+    Type: 'RoundCreated';
+    roundParameters: {
+        network: string;
+        miningFeeRate: number;
+        coordinationFeeRate: {
+            rate: number;
+            plebsDontPayThreshold: number;
+        };
+        maxSuggestedAmount: number;
+        minInputCountByRound: number;
+        maxInputCountByRound: number;
+        allowedInputAmounts: AllowedRange;
+        allowedOutputAmounts: AllowedRange;
+        allowedInputScriptTypes: AllowedScriptTypes[];
+        allowedOutputScriptTypes: AllowedScriptTypes[];
+        standardInputRegistrationTimeout: string;
+        connectionConfirmationTimeout: string;
+        outputRegistrationTimeout: string;
+        transactionSigningTimeout: string;
+        blameInputRegistrationTimeout: string;
+        minAmountCredentialValue: number;
+        maxAmountCredentialValue: number;
+        initialInputVsizeAllocation: number;
+        maxVsizeCredentialValue: number;
+        maxVsizeAllocationPerAlice: number;
+        maxTransactionSize: number;
+        minRelayTxFee: number;
+    };
+}
+
+export interface CoinjoinInput {
+    outpoint: string;
+    txOut: {
+        scriptPubKey: string; // format: "0 f23290d9f9be3d13a315b6febe29fc0786d34c96"
+        value: number;
+    };
+    ownershipProof: string;
+}
+
+export interface CoinjoinInputAddedEvent {
+    Type: 'InputAdded';
+    coin: CoinjoinInput;
+}
+
+export interface CoinjoinOutput {
+    scriptPubKey: string; // format: "0 76215d74689b52e41c1636e46df04bde793be57a"
+    value: number;
+}
+
+export interface CoinjoinOutputAddedEvent {
+    Type: 'OutputAdded';
+    output: CoinjoinOutput;
+}
+
+export type CoinjoinStateEvent =
+    | CoinjoinRoundCreatedEvent
+    | CoinjoinInputAddedEvent
+    | CoinjoinOutputAddedEvent;
+
+export interface AllowedRange {
+    min: number;
+    max: number;
+}
+export type AllowedScriptTypes = 'P2WPKH' | 'Taproot';
+
+export interface CoinjoinState {
+    Type: string; // TODO enum?
+    events: CoinjoinStateEvent[];
+    isFullySigned?: boolean;
+    witnesses?: Record<number, string>;
+}
+
+export enum RoundPhase {
+    InputRegistration = 0,
```

Comment:
nit: what is the point of setting the those initialization values explicitely? enums are numeric by default